src/main/java/com/palominolabs/metrics/guice/provider/AnnotationProvider.java

```diff
@@ -0,0 +1,33 @@
+package com.palominolabs.metrics.guice.provider;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+
+import javax.annotation.Nullable;
+
+public class AnnotationProvider {
```

Comment:
I like making this more configurable for users, but I think this abstraction can be improved by making it less specific: if we're letting people configure how methods get intercepted beyond only doing the default "the method must be annotated" logic, we might as well allow more flexibility than just type or method. If we have something like a `MethodMatcher` interface, we could ship a default one that uses annotations on methods, another that allows annotations on classes also, and let people provide their own implementation that does anything they want.